File: app/src/main/java/com/jesta/util/Jesta.java
package com.jesta.util;

import java.util.HashMap;

public class Jesta {


    private String id;
    private String difficulty;
    private String description;
    private String imageUrl;
    private long payment;
    private long numOfPeople;
    private long duration;
    private String location;

    public Jesta(String id, String difficulty, String description, String imageUrl, int payment, int numOfPeople, int duration, String location) {
        this.id = difficulty;
        this.difficulty = difficulty;
        this.description = description;
        this.imageUrl = imageUrl;
        this.payment = payment;
        this.numOfPeople = numOfPeople;
        this.duration = duration;
        this.location = location;
    }

    public Jesta(HashMap dbJesta) {
        this.id = (String)dbJesta.get("id");
        this.difficulty = (String)dbJesta.get("difficulty");
        this.description = (String)dbJesta.get("description");
        this.imageUrl = (String)dbJesta.get("imageUrl");
        this.payment = (long) dbJesta.get("payment");
        this.numOfPeople = (long)dbJesta.get("numOfPeople");
        this.duration = (long)dbJesta.get("duration");
        this.location = (String)dbJesta.get("location");
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(String difficulty) {
        this.difficulty = difficulty;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public long getPayment() {
        return payment;
    }

    public void setPayment(int payment) {
        this.payment = payment;
    }

    public long getNumOfPeople() {
        return numOfPeople;
    }

    public void setNumOfPeople(int numOfPeople) {
        this.numOfPeople = numOfPeople;
    }

    public long getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }



}

File: app/src/main/java/com/jesta/util/Jesta.kt
package com.jesta.util

import android.os.Parcelable
import com.google.firebase.auth.FirebaseUser
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Mission(
    var jestaId: String = "emptyJestaID",
    var authorId: String = "emptyAuthorID",
    var title: String = "emptyTitle",
    var difficulty: String = "emptyDifficulty",
    var description: String = "emptyDescription",
    var imageUrl: String = "emptyImageUrl",
    var payment: Int = -1,
    var numOfPeople: Int = -1,
    var duration: Int = -1,
    var location: String = "emptyLocation",
    var diamonds: Int = -1,
    var tags: List<String> = listOf("Tag1", "Tag2")
) : Parcelable {

    constructor(dbJesta: Map<String, String>) : this() {
        jestaId = dbJesta["jestaId"].toString()
        authorId = dbJesta["authorId"].toString()
        difficulty = dbJesta["difficulty"].toString()
        description = dbJesta["description"].toString()
        imageUrl = dbJesta["imageUrl"].toString()
        payment = dbJesta["payment"].toString().toInt()
        numOfPeople = dbJesta["numOfPeople"].toString().toInt()
        duration = dbJesta["duration"].toString().toInt()
        location = dbJesta["location"].toString()

        // TODO add diamonds prop
        diamonds = dbJesta["diamonds"].toString().toInt()
        title = dbJesta["title"].toString()

        // TODO write initializer from JSON
//        tags = dbJesta["tags"]
    }

}

// TODO: tweak to use Post and not Mission
//data class Post(val user: User, val mission: Mission) {
//}

data class User(
    var id: String = "emptyID",
    var email: String = "emptyEmail",
    var displayName: String = "emptyName",
    var photoUrl: String = "emptyPhoto",
    var phoneNumber: String = "emptyNumber"
) {

    constructor(firebaseUser: FirebaseUser) : this() {
        id = firebaseUser.uid
        email = firebaseUser.email.toString()
        photoUrl = firebaseUser.photoUrl.toString()
        displayName = firebaseUser.displayName.toString()
        phoneNumber = firebaseUser.phoneNumber.toString()
    }

    constructor(dbUser: Map<String, String>) : this() {
        id = dbUser["id"].toString()
        email = dbUser["email"].toString()
        photoUrl = dbUser["photoUrl"].toString()
        displayName = dbUser["displayName"].toString()
        phoneNumber = dbUser["phoneNumber"].toString()
    }
}
